src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1193,6 +1193,12 @@ private void CheckAllowedProjectChanges(ProjectChanges projectChanges)
                 throw new NotSupportedException(WorkspacesResources.Changing_compilation_options_is_not_supported);
             }
 
+            if (projectChanges.OldProject.DefaultNamespace != projectChanges.NewProject.DefaultNamespace
```

Comment:
What about other properties being checked here? If you changed something else we still won't apply it?